src/ui/public/visualize/visualize.js

```diff
@@ -159,38 +153,30 @@ uiModules
             if ($scope.vis.type.requiresSearch && ['query', 'filters'].includes(keys[0])) {
               $scope.fetch();
             }
-            if (keys[0] === 'uiState') {
```

Comment:
Instead of watching the `uiState` we earlier watched (if available) the `appState` and checked whether it's `uiState` key changed to refresh.

This would always assume, that the current `uiState` was stored with the key `uiState` in the `appState`. Since more recent dashboard refactoring to use embeddableConfigs, this wasn't the case anymore, and `visualize` would never pick up changes to the `uiState`, since the key that actually would change was now named `embeddableConfig`.

This code is now replaced, by a regular watch on the `uiState` instead of trying to make assumptions where it's stored in the `appState`.